config/opal_check_libnl.m4

```diff
@@ -0,0 +1,228 @@
+dnl -*- shell-script -*-
+dnl
+dnl Copyright (c) 2015-2016 Research Organization for Information Science
+dnl                         and Technology (RIST). All rights reserved.
+dnl $COPYRIGHT$
+dnl
+dnl Additional copyrights may follow
+dnl
+dnl $HEADER$
+dnl
+
+dnl
+dnl Helper function to parse --with-libnl* options
+dnl
+dnl $1: variable name
+dnl $2: --with-<foo> value
+dnl
+AC_DEFUN([OPAL_LIBNL_PARSE_WITH],[
+	case "$2" in
+	no)
+		# Nope, don't want it
+		opal_want_$1=no
+		;;
+	yes)
+		# Yes, definitely want it
+		opal_want_$1=yes
+		;;
+	default)
+		# Default case -- try and see if we can find it
+		opal_want_$1=default
+		opal_$1_location=/usr
+		;;
+	*)
+		# Yes, definitely want it -- at a specific location
+		opal_want_$1=yes
+		opal_$1_location="$2"
+		;;
+	esac
+])
+
+dnl OPAL_LIBNL_SANITY_INIT()
+dnl --------------------------------------------------------------------
+AC_DEFUN([OPAL_LIBNL_SANITY_INIT], [
+    opal_libnl_version=0
+    opal_libnl_libs=
+    opal_libnl3_libs=
+    AC_ARG_WITH([libnl],
```

Comment:
I find reading this patch to be a bit confusing -- because we use the string "libnl" to both mean "any version of the netlink library" and "version 1 of the netlink library."

Perhaps we should use this nomenclature:

- `libnl`: a version-agnostic way of referring to the netlink library
- `libnlv1` (as a variable, or `libnl v1` as prettyprint): refers to version 1 of the netlink library
- `libnlv3` (as a variable, or `libnl v3 as prettyprint): refers to version 3 of the netlink library

This makes it clear that `--with-libnl` refers to either version of the netlink library.  But all macros and variables are clear in whether they are dealing with v1 or v3.